include/mlir/IR/OpImplementation.h

```diff
@@ -655,6 +656,13 @@ class OpAsmDialectInterface
   /// OpAsmInterface.td#getAsmResultNames for usage details and documentation.
   virtual void getAsmResultNames(Operation *op,
                                  OpAsmSetValueNameFn setNameFn) const {}
+
+  /// Get a special name to use when printing the entry block arguments of the
+  /// region contained by an operation in this dialect.  The desired name should
+  /// be streamed into 'os'.  If the result is empty, the default name will be
+  /// used.
+  virtual void getRegionArgumentName(BlockArgument *arg,
+                                     raw_ostream &os) const {}
```

Comment:
Can we set this up with the same structure as the result names? i.e. allow groups of names. That would make things more consistent and extensible.

Reply:
It would also be nice to add a hook to the OpAsmOpInterface, but that requires support for default implementations which is something I need to add. So I can do that as a follow up.

Reply:
It's not clear to me how to group block argument names, but I did modify the hook to be more consistent with `getAsmResultNames`